From 2a619112843f29e1926023c6bc612216064ed4f0 Mon Sep 17 00:00:00 2001
From: Kazunori Kobayashi <kkobayas@igel.co.jp>
Date: Tue, 14 May 2013 15:36:13 +0900
Subject: [PATCH 2/2] drmkms: fix framebuffer allocation size

config.size.h has the real height of an image, so framebuffer size
calculation for packed color formats is done successfully.
However the framebuffer size in planar or semi-planar formats is
set incorrectly because the chroma plane size isn't added to the size.
This patch multiplies the pitch by drm_fake_height, which is takes into
account the size of the chroma plane.

Signed-off-by: Kazunori Kobayashi <kkobayas@igel.co.jp>
---
 systems/drmkms/drmkms_surface_pool.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/systems/drmkms/drmkms_surface_pool.c b/systems/drmkms/drmkms_surface_pool.c
index fe01800..28bad69 100644
--- a/systems/drmkms/drmkms_surface_pool.c
+++ b/systems/drmkms/drmkms_surface_pool.c
@@ -391,7 +391,7 @@ drmkmsAllocateBuffer( CoreSurfacePool       *pool,
      kms_bo_get_prop(alloc->bo, KMS_PITCH, &alloc->pitch);
 #endif
 
-     alloc->size = alloc->pitch * surface->config.size.h;
+     alloc->size = alloc->pitch * drm_fake_height;
 
      D_DEBUG_AT( DRMKMS_Surfaces, "  -> bo        %p\n", alloc->bo );
      D_DEBUG_AT( DRMKMS_Surfaces, "  -> bo.handle %u\n", alloc->handle );
-- 
1.8.1.2

